src/main/java/org/candlepin/service/ProductServiceAdapter.java

```diff
@@ -50,6 +51,17 @@ public interface ProductServiceAdapter {
      */
     Collection<? extends ProductInfo> getProductsByIds(String ownerKey, Collection<String> ids);
 
+    /**
+     * Retrieves the products matching the provided top level SKU IDs. Only the found products will be
+     * returned. When no matching products are found, an empty list will be returned.
+     *
+     * @param skuIds
+     *  the top level SKU IDs to fetch products for
+     *
+     * @return the products that match the provided SKU IDs
+     */
+    List<ProductInfo> getProductsBySkuIds(Collection<String> skuIds);
```

Comment:
We may want to call this something more generic like "get children by product IDs" or some such. That we're passing in a "SKU" and getting engineering products back out is kind of an implementation detail based on how products are sold today -- it's not necessarily true going forward. And in the case of some future world where n-tier products must be supported, the term "SKU" loses meaning entirely.

Reply:
Yeah, that makes sense! Updated